Assets/Mirror/Runtime/NetworkDiagnostics.cs

```diff
@@ -48,7 +48,7 @@ internal MessageInfo(IMessageBase message, int channel, int bytes, int count)
         /// </summary>
         public static event Action<MessageInfo> OutMessageEvent;
 
-        internal static void OnSend<T>(T message, int channel, int bytes, int count) where T : IMessageBase
+        internal static void OnSend<T>(T message, int channel, int bytes, int count) where T : INetworkMessage
```

Comment:
I think the NetworkProfiler will need updating for this change

Reply:
Yes,  it will